### PR TITLE
Fix rendering of children

### PR DESCRIPTION
**Pull request by spydon**
# Description

Since the canvas state isn't saved at the correct place the children weren't rendered properly.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on the latest `master`
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/components/base_component.dart [hunk 1]
```diff
@@ -56,13 +56,12 @@ abstract class BaseComponent extends Component {
...
   @override
   void render(Canvas canvas) {
+    canvas.save();
     prepareCanvas(canvas);
 
     if (debugMode) {
       renderDebugMode(canvas);
     }
-
-    canvas.save();
```
Comment (spydon) [inline lib/components/base_component.dart:65]: Maybe we should have an optional argument for `BaseComponent` whether children should be rendered in relation to the parent or not?

**Review by erickzanardo — APPROVED** [review]

**Comment by erickzanardo** [comment]
I wonder if we can have an example which showcases this, what do you think?

File: lib/components/base_component.dart [hunk 2]
```diff
@@ -56,13 +56,12 @@ abstract class BaseComponent extends Component {
...
     _children.forEach((c) => c.render(canvas));
     canvas.restore();
```
Comment (spydon) [inline lib/components/base_component.dart:66]: Wait, this is wrong... Then the whole preparation of the canvas will be in vain.

**Comment by spydon** [comment]
> I wonder if we can have an example which showcases this, what do you think?

Added an example in #579

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**